packages/compiler/test/test-parser-utils.ts

```diff
@@ -0,0 +1,73 @@
+import { ok, strictEqual } from "assert";
+import {
+  CadlScriptNode,
+  IdentifierNode,
+  NamespaceStatementNode,
+  SyntaxKind,
+} from "../core/index.js";
+import { getNodeAtPosition, parse } from "../core/parser.js";
+import { Node } from "../core/types.js";
+import { extractCursor } from "./server/test-server-host.js";
+
+describe("compiler: parser utils", () => {
+  describe("getNodeAtPosition", () => {
+    async function getNodeAtCursor(
+      sourceWithCursor: string
+    ): Promise<{ root: CadlScriptNode; node: Node | undefined }> {
+      const { source, pos } = extractCursor(sourceWithCursor);
+      const root = parse(source);
+      return { node: getNodeAtPosition(root, pos), root };
+    }
+
+    it("return namespace when in namespace body", async () => {
+      const { node } = await getNodeAtCursor(`
+        namespace Foo {
+          ┆
+        }
+      `);
+      ok(node);
+      strictEqual(node.kind, SyntaxKind.NamespaceStatement);
+      strictEqual((node as NamespaceStatementNode).id.sv, "Foo");
```

Comment:
IIRC, you can remove these casts because strictEqual `asserts` allow TS to narrow type.